truffle/src/main/java/org/jruby/truffle/nodes/StringCachingGuards.java

```diff
@@ -13,33 +13,37 @@
 import com.oracle.truffle.api.object.DynamicObject;
 import org.jruby.truffle.runtime.core.StringOperations;
 import org.jruby.truffle.runtime.layouts.Layouts;
+import org.jruby.truffle.runtime.rope.Rope;
 import org.jruby.util.ByteList;
 
 public abstract class StringCachingGuards {
 
-    public static ByteList privatizeByteList(DynamicObject string) {
+    public static Rope privatizeRope(DynamicObject string) {
         if (RubyGuards.isRubyString(string)) {
-            return StringOperations.getByteListReadOnly(string).dup();
+            // TODO (nirvdrum 25-Jan-16) Should we flatten the rope to avoid caching a potentially deep rope tree?
```

Comment:
Indeed. Although we're still doing a full byte[] comparison if the ropes aren't identity equal. We can eliminate that, but I was just maintaining existing functionality for now. The flattening could also work in conjunction with a lookup table to ensure the same references, as well.